game.py

```diff
@@ -0,0 +1,302 @@
+            if beta <= alpha:
+                print ('Break min')
+                break
+            
+        return best_move, beta
+    
+def get_computer_move(game_state):
+    player_next_first = game_state.get_player_next()
+    look_forward = choose_look_ahead()
+    sort_depth = get_sort_depth()
+    score_count = 0
+    best_move_and_score = minimax(game_state, look_forward, player_next_first, -inf, inf, sort_depth)
+    return best_move_and_score
+
+
+def choose_board_size():
+    rows = int(input("Enter number of rows : "))
+    columns = int(input("Enter number of columns : "))
+    return (rows, columns)
+
+def create_board(board_size):
+    board = [[None] * int(board_size[1]) for _ in range(int(board_size[0]))]
+    return board
+
+def choose_look_ahead():
+    look_ahead = int(input("Enter look ahead : "))
+    return look_ahead
+
+def move_more():
+    more_moves = input("Another Move? (Y/N) : ")
+    if more_moves == 'Y' or more_moves == 'y':
+        return True
+    else:
+        return False
+
+def undo_move():
+    undo_last_move = input("Undo last move? (Y/N) : ")
+    if undo_last_move == 'Y' or undo_last_move == 'y':
+        return True
+    else:
+        return False
+
+def get_sort_depth():
+    sort_depth = int(input("Enter sort depth : "))
+    return sort_depth
+
+def human_to_play():
+    human = input("Human to play? (Y/N) : ")
+    if human == 'Y' or human == 'y':
+        return True
+    else:
+        return False
+
+def get_human_move(game_state):
+    valid = False
+    while not valid:
+        row = int(input("Enter row : ")) - 1
+        column = int(input("Enter column : ")) - 1
+        if  game_state.valid_move((row, column)):
+            valid = True
+        else:
+            print('Invalid move. Try again')
+            valid = False
+    return row, column
+
+def get_hint():
+    hint = input("Get hint? (Y/N) : ")
+    if hint == 'Y' or hint == 'y':
+        return True
+    else:
+        return False
+
+
+def main():
+    board = []
+    board_size = choose_board_size()
+    print('board size : ', board_size)
+    board = create_board(board_size)
+    print('board : ', board)
+    
+    player_next = 1
+    number_of_moves = 0
+    
+    global score_count
+    
+    game_state = GameState(board, player_next)
+   
+    another_move = True
+
+    while another_move:
+
+        if  human_to_play():
+            while get_hint():
+                score_count = 0
+                best_move_and_score = get_computer_move(game_state)
+                best_move = best_move_and_score[0]
+                print('Suggested move row : ', best_move[0] + 1, 'column : ', best_move[1] + 1)
+            
+            move = get_human_move(game_state)
+            
+            game_state = game_state.make_move(move)
+            number_of_moves = number_of_moves + 1 
+            print('board = ', game_state.get_board(), 'next player = ', game_state.get_player_next())

```

Comment:
Playing the game, my biggest gripe is with the way we are printing the board. I understand it's a prototype and we're about to try and move out of the command-line with the display, but yea it's extremely hard to tell what's going on at the moment. Figuring out my score in my head from the print-out on even a 3x4 board is beyond my capabilities. I'm surprised you were able to debug/etc with it like this(!)

Might be a good exercise to try and build a multi-line string from the board list-of-lists. The one gotcha is the inversion of the y-axis for screen coordinates vs. cartesian.